File: DSA/LeetCode/Arrays/9 - Palindrome Number/Solution.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < 0) {
    return false
  }

  let num = x
  let palin = 0

  while (num > 0) {
    const r = num % 10
    palin = (palin * 10) + r
    num = parseInt(num / 10)
  }
  return x === palin
};

File: DSA/LeetCode/Arrays/9 - Palindrome Number/Solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        palin, num = 0, x
        while x > 0:
            palin = (palin*10) + x % 10
            x //= 10

        return num == palin
